sdk/src/record-scanner.ts

```diff
@@ -0,0 +1,362 @@
+                }),
+            );
+
+            return await response.json();
+        } catch (error) {
+            console.error(`Failed to check if serial numbers exist: ${error}`);
+            throw error;
+        }
+    }
+
+    /**
+     * Check if a list of tags exist in the record scanner service.
+     * 
+     * @param {string[]} tags The tags to check.
+     * @returns {Promise<Record<string, boolean>>} Map of Aleo Record tags and whether they appeared in any inputs on chain. If boolean corresponding to the tag has a true value, that Record is considered spent by the Aleo Network.
+     */
+    async checkTags(tags: string[]): Promise<Record<string, boolean>> {
+        try {
+            const response = await this.request(
+                new Request(`${this.url}/records/tags`, {
+                    method: "POST",
+                    headers: { "Content-Type": "application/json" },
+                    body: JSON.stringify(tags),
+                }),
+            );
+            
+            return await response.json();
+        } catch (error) {
+            console.error(`Failed to check if tags exist: ${error}`);
+            throw error;
+        }
+    }
+
+    /**
+     * Check the status of a record scanner indexing job.
+     * 
+     * @param {string} jobId The job id to check.
+     * @returns {Promise<StatusResponse>} The status of the job.
+     */
+    async checkStatus(jobId: string): Promise<StatusResponse> {
+        try {
+            const response = await this.request(
+                new Request(`${this.url}/status`, {
+                    method: "POST",
+                    headers: { "Content-Type": "application/json" },
+                    body: JSON.stringify(jobId),
+                }),
+            );
+
+            return await response.json();
+        } catch (error) {
+            console.error(`Failed to check status of job: ${error}`);
+            throw error;
+        }
+    }
+
+    /**
+     * Find a record in the record scanner service.
+     * 
+     * @param {OwnedFilter} searchParameters The filter to use to find the record.
+     * @returns {Promise<OwnedRecord>} The record.
+     */
+    async findRecord(searchParameters: OwnedFilter): Promise<OwnedRecord> {
+        try {
+            const records = await this.findRecords(searchParameters);
+
+            if (records.length > 0) {
+                return records[0];
+            }
+
+            throw new Error("Record not found");
+        } catch (error) {
+            console.error(`Failed to find record: ${error}`);
+            throw error;
+        }
+    }
+
+    /**
+     * Find records in the record scanner service.
+     * 
+     * @param {OwnedFilter} filter The filter to use to find the records.
+     * @returns {Promise<OwnedRecord[]>} The records.
+     */
+    async findRecords(filter: OwnedFilter): Promise<OwnedRecord[]> {
+        if (!this.uuid) {
+            throw new Error("Not registered");
```

Comment:
We should strive to have a Rust compiler level of informative errors so that users get good guidance for the next thing they should do if they experience an error.

```suggestion
            throw new Error("You are using the RecordScanner implementation of the RecordProvider. No account has been registered with the RecordScanner which is required to use the findRecords method. Please set an with the setAccount method before calling the findRecords method again.");
```

Reply:
done